src/org/daisy/dotify/hyphenator/impl/SimpleHyphenator.java

```diff
@@ -0,0 +1,322 @@
+    private int rightHyphenMin = 1;
+
+    /*
+    Lastly we have a hyphenation cache so we don't hyphenate a word more than once.
+     */
+    private final Map<String, String> hyphCache = new HashMap<>();
+
+    /**
+     * The constructor loads the dictionary with all the words package in a hunspell / LaTex similar format.
+     *
+     * @param locale  Locale of the language to load. We have specified all the available languages and their
+     *                dictionary files within SimpleHyphenator.properties.
+     *
+     * @throws HyphenatorConfigurationException If we aren't able to read the bundle or find / load the dictionary
+     *                this will throw a configuration exception.
+     */
+    SimpleHyphenator(String locale) throws HyphenatorConfigurationException {
+        try {
+            ResourceBundle p = PropertyResourceBundle.getBundle("org/daisy/dotify/hyphenator/impl/SimpleHyphenator");
+            String patternPath = "resource-files/dict/" + p.getString(locale);
+
+            InputStream language = SimpleHyphenator.getResource(patternPath).openStream();
+            InputStreamReader sr = new InputStreamReader(language, StandardCharsets.UTF_8);
+
+            loadTable(sr);
+        } catch (Exception e) {
+            throw new SimpleHyphenatorConfigurationException(e);
+        }
+    }
+
+    /**
+     * The load table function reads the dictionary into hyphenWords. First we ignore all comments and then
+     * we look for the left min hyphen and right min hyphen so we could use that configuration from the header.
+     * Currently we ignore the encoding and compound words part of the configuration. All files expects to be
+     * in a UTF-8 encoding and until we get a dictinary where compound words are used we will not use that
+     * configuration.
+     *
+     * @param isr   The stream reader used to read dictionary file.
+     *
+     * @throws IOException  If we can't find or read the file the method will throw an exception.
+     */
+    public void loadTable(InputStreamReader isr) throws IOException {
+        BufferedReader br = new BufferedReader(isr);
+        String line;
+        while ((line = br.readLine()) != null) {
+            if (line.contains("%")) {
+                line = line.substring(0, line.indexOf("%"));
+            }
+            if (
+                line.isEmpty() ||
+                line.trim().equals("") ||
+                line.toUpperCase().contains("COMPOUNDLEFTHYPHENMIN") ||
+                line.toUpperCase().contains("COMPOUNDRIGHTHYPHENMIN") ||
+                line.toUpperCase().contains("ISO8859-1") ||
+                line.toUpperCase().contains("UTF-8")
+            ) {
+                continue;
+            }
+            if (line.toUpperCase().startsWith("LEFTHYPHENMIN")) {
+                leftHyphenMin = Integer.parseInt(line.split(" ")[1]);
+                continue;
+            }
+            if (line.toUpperCase().startsWith("RIGHTHYPHENMIN")) {
```

Comment:
These words can only appear at the start of the file, so some optimization is possible here.